src/cobra/manipulation/delete.py

```diff
@@ -57,113 +61,43 @@ def prune_unused_reactions(model: "Model") -> Tuple["Model", List["Reaction"]]:
     return output_model, reactions_to_prune
 
 
-def undelete_model_genes(model: "Model") -> None:
-    """Undo the effects of a call to `delete_model_genes` in place.
-
-    Parameters
-    ----------
-    model: cobra.Model
-        The model which will be modified in place.
-
-    """
-    if model._trimmed_genes is not None:
-        for x in model._trimmed_genes:
-            x.functional = True
-
-    if model._trimmed_reactions is not None:
-        for (
-            the_reaction,
-            (lower_bound, upper_bound),
-        ) in model._trimmed_reactions.items():
-            the_reaction.lower_bound = lower_bound
-            the_reaction.upper_bound = upper_bound
-
-    model._trimmed_genes = []
-    model._trimmed_reactions = {}
-    model._trimmed = False
-
-
-def get_compiled_gene_reaction_rules(model: "Model") -> Dict["Reaction", Module]:
-    """Generate a dictionary of compiled gene-reaction rules.
-
-    Any gene-reaction rule expression which cannot be compiled or do not
-    evaluate after compiling will be excluded. The result can be used in the
-    `find_gene_knockout_reactions` function to speed up evaluation of these
-    rules.
-
-    Parameters
-    ----------
-    model: cobra.Model
-        The model to get gene-reaction rules for.
-
-    Returns
-    -------
-    dict of cobra.Reaction, ast.Module
-        The dictionary of cobra.Reaction objects as keys and ast.Module
-        objects as keys.
-
-    .. deprecated::
-        Internal function that has outlived its purpose.
-
-    """
-    warn(
-        "The function `get_compiled_gene_reaction_rules` has outlived its purpose. "
-        "It will be removed soon.",
-        DeprecationWarning,
-    )
-    return {r: r.gpr for r in model.reactions}
-
-
-def find_gene_knockout_reactions(
+def knock_out_model_genes(
     model: "Model",
-    gene_list: List["Gene"],
-    compiled_gene_reaction_rules: Optional[Dict["Reaction", Module]] = None,
+    gene_list: Union[List["Gene"], Set["Gene"], List[str], Set[str]],
```

Comment:
When using `get_by_any`:

```suggestion
    gene_list: Iterable[Union["Gene", int, str]],
```